src/libostree/ostree-impl-system-generator.c

```diff
@@ -289,11 +289,18 @@ _ostree_impl_system_generator (const char *normal_dir, const char *early_dir, co
   if (unlinkat (AT_FDCWD, INITRAMFS_MOUNT_VAR, 0) == 0)
     return TRUE;
 
+#ifdef OSTREE_PREPARE_ROOT_STATIC
+  // Create /run/ostree-booted now, because other things rely on it.
+  // If the system compiled with a static prepareroot, then our generator
+  // makes a hard assumption that ostree is in use.
+  touch_run_ostree ();
```

Comment:
If the touch fails we don't log anything and we just continue running

Reply:
True, but this was also the case before; and there's even a comment in the source code that that's intentional. I think the only real thing that could fail here is ENOSPC but if you hit that this early on with `/run` the system is going to fall over in a ton of ways regardless.

I don't mind changing this of course, but not sure about doing it in this PR.

Reply:
I was thinking if /run is not created yet, or read-only because we have not mounted a tmpfs on top yet
I would at least log an error to help with debugging in any cases, but this can be done in another PR